dumbo/src/tcp/connection.rs

```diff
@@ -1258,7 +1258,7 @@ pub(crate) mod tests {
     }
 
     #[test]
-    #[allow(clippy::cyclomatic_complexity)]
+    #[allow(clippy::cognitive_complexity)]
```

Comment:
nit: this test is actually difficult to understand.

Reply:
We should open a separate issue for this particular problem.